### PR TITLE
add compression in agg_layer.

### PR DESCRIPTION
**Pull request by fy222fy**
# Preamble

For Pull Request initiator, please read the following steps carefully. 请合并请求的发起人仔细阅读下列步骤。

---

If you are creating a `Pull Request` about listed topic below, please go the the `Preview` tab and click the relative link:
如果您要创建一个有关下列内容的的合并请求，请切换到 `Preview` 面板并点击下方相关的链接:

* Select [Awesome-PETs](?expand=1&template=pull_request_template_pets.md) for files under docs/awesome-pets

---

Select a proper `Pull Request` template can accelarate the review process, thanks for your cooperation.
选择一个正确的`合并请求`模版可以加速审查流程，谢谢您的配合

If your `Pull Request` topic is not amoung the list above, please just delete all prepopulated template content.
如果您的`合并请求`内容不在上述的列表里，请删除提前填入的模版内容。

---

Thanks for your contribution to SecretFlow project
感谢您对于 SecretFlow 项目的贡献

### CODE REVIEW

**Comment by fy222fy** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by 6fj** [comment]
recheck

File: secretflow/ml/nn/sl/agglayer/agg_layer.py [hunk 1]
```diff
@@ -145,13 +145,9 @@ def get_reg_loss(f_data: ForwardData):
 
     @staticmethod
     def set_forward_data(
-        hidden,
-        losses,
+        hidden, losses,
```
Comment (zhangxingmeng) [inline secretflow/ml/nn/sl/agglayer/agg_layer.py:148]: format不对，需要用black -S处理一下

File: secretflow/ml/nn/sl/agglayer/agg_layer.py [hunk 2]
```diff
@@ -246,17 +244,19 @@ def split_to_parties(self, data: List) -> List[PYUObject]:
...
             data_slice = data[start_idx : start_idx + self.basenet_output_num[p]]
-
             result.append(data_slice)
             start_idx = start_idx + self.basenet_output_num[p]
         return result
 
     def collect(self, data: Dict[PYU, DeviceObject]) -> List[DeviceObject]:
-        """Collect data from participates"""
+        """Collect data from participates
+        todo: Support compress communication when using agg method.
```
Comment (zhangxingmeng) [inline secretflow/ml/nn/sl/agglayer/agg_layer.py:258]: todo->TODO

File: secretflow/ml/nn/sl/agglayer/agg_layer.py [hunk 3]
```diff
@@ -318,10 +323,7 @@ def scatter(self, data: Union[List, PYUObject]) -> Dict[PYU, DeviceObject]:
         return result
 
     def forward(
-        self,
-        data: Dict[PYU, DeviceObject],
-        axis=0,
-        weights=None,
+        self, data: Dict[PYU, DeviceObject], axis=0, weights=None,
```
Comment (zhangxingmeng) [inline secretflow/ml/nn/sl/agglayer/agg_layer.py:326]: ditto,format问题

File: secretflow/ml/nn/sl/agglayer/agg_layer.py [hunk 4]
```diff
@@ -370,23 +365,29 @@ def forward(
...
-                self.fuse_sparse_masks = fuse_sparse_masks
-                self.is_compressed = is_compressed
-
-            return data
+                (
+                    compute_data,
+                    self.fuse_sparse_masks,
+                    self.is_compressed,
+                ) = self.device_y(
+                    self.handle_sparse_hiddens, num_returns=3,  # 解压
```
Comment (zhangxingmeng) [inline secretflow/ml/nn/sl/agglayer/agg_layer.py:383]: format问题

**Review by zhangxingmeng — APPROVED** [review]
LGTM

**Pull request merged**